src/DataflowPass.cpp

```diff
@@ -168,79 +205,132 @@ void DataflowPass::transfer(Instruction *instruction,
       bool loopBroken = false;
       for (auto Pair : this->memoryFunctions) {
         if (fnName == Pair.first) {
-          logout("not calling onalloc for "
-                 << arg << " of fnName " << fnName << " and inst "
-                 << *instruction << " and just returning now") return;
+          logout("not calling onalloc for " << arg << " of fnName " << fnName
+                                            << " and inst " << *instruction
+                                            << " and just returning now");
+          return;
 
         } else if (fnName == Pair.second) {
-          MethodsSet *methods =
-              inputProgramPoint.getPVRef(arg, false)->getMethodsSetRef();
-          logout("GETMAIN-5 ON '"
-                 << arg
-                 << "'") this->onDeallocationFunctionCall(methods, fnName);
+          MethodsSet *methods = pv->getMethodsSetRef();
+          logout("GETMAIN-5 ON '" << arg << "'");
+          this->onDeallocationFunctionCall(methods, fnName);
           loopBroken = true;
           break;
         }
       }
 
-      if (loopBroken) {
+      if (loopBroken || fnName == "llvm.dbg.declare") {
         continue;
       }
 
-      if (fnName == "llvm.dbg.declare") {
-        continue;
+      // finds annotations
+      logout("finding annotations for " << fnName);
+
+      /*
+        LLVM uses sret attributes as an optimization if the return type is
+        large. a struct with more than 2 fields can trigger this optimization.
+
+        docs: https://llvm.org/docs/LangRef.html#parameter-attributes
+        less credible forum post:
+        https://discourse.llvm.org/t/optimizing-sret-on-caller-side/60660
+
+        our instructions end up looking like this:
+        call void @does_something_a(%struct.my_struct* sret %A1,
+        %struct.my_struct* byval align 8 %A), !dbg !58
+
+        here's how we will handle these types of instructions:
+
+        say we have the instruction
+        call void @does_something_a(%struct.my_struct* sret %A1,
+        %struct.my_struct* byval align 8 %A), !dbg !58
+
+        (this is from C code
+        my_struct A1 = does_something_a(A);)
+
+        *we assume LHS is always a struct and first argument always has sret
+        attribute
+
+        we will:
+        -destructure A1 into its fields (A1.0, A1.1, ...)
+        -determine A1's MustCall/CalledMethods from annotations on return on
+        does_something_a -destructure A into its fields (A.0, A.1, ...)
```

Comment:
I think there's a missing newline in this comment. Otherwise, it doesn't really make sense.